### PR TITLE
Update article about Source Set Hierarchies for 1.9.20

### PR DESCRIPTION
**Pull request by dsavvinov**
See https://youtrack.jetbrains.com/issue/KT-60843/Finalize-public-communication-about-hierarchy-templates

### CODE REVIEW

**Review by danil-pavlov — APPROVED** [review]

File: docs/topics/multiplatform/multiplatform-hierarchy.md [hunk 1]
```diff
@@ -116,114 +131,106 @@ to your gradle.properties
...
-#### Replacing your manual configuration with the Default Hierarchy Template
-
-**Case.** All added dependsOn-edges are covered by the Default Hierarchy Template. This can happen, for example, if 
-you're migrating previously existing project to 1.9.20 or if you have used some code snippets written before 1.9.20.
+To solve this issue, configure your project by doing one of the following:
 
-**Solution.** Remove all the manual dependsOn-edges (`dependsOn`-calls) and instantiation of the intermediate source sets
-(`by creating`-calls). 
+* [Replace your manual configuration with the default hierarchy template](#replacing-manual-configuration)
+* [Add your source sets to the default hierarchy template](#adding-source-sets)
```
Comment (dsavvinov) [inline docs/topics/multiplatform/multiplatform-hierarchy.md:137]: Please don't use this wording. "Adding source sets to the default hierarchy template" 100% reads as a completely different action that actually *modified* the default hierarchy template. That is a frequent request, but it's explicitly forbidden for now. 

What we do here instead is that we create additional source sets *alongside* the ones created by the template. 

Please reformulate

File: docs/topics/multiplatform/multiplatform-hierarchy.md [hunk 2]
```diff
@@ -116,114 +131,106 @@ to your gradle.properties
...
-#### Modifying the source sets created by the Default Hierarchy Template
+**Case**. You already have the source sets with the exact same names as those generated by the template but shared among
+different sets of targets in your project. For example, a `nativeMain` source set shared among not all the native 
+targets but only the desktop ones: `linuxX64`, `mingwX64`, and `macosX64`.
 
-**Case.** You want to have a source set exactly with the same name as the one created by the Default Target Hierarchy,
-but shared for a different set of targets. For example, you want to have a `nativeMain` shared not for all native 
-targets, but only for the desktop ones: `linuxX64`, `mingwX64`, `macosX64`.
+**Solution**. There's currently no way to modify the default `dependsOn` relations among the template's source sets.
+It's a technical limitation of the template. It's also important that the implementation and the meaning of source sets,
```
Comment (dsavvinov) [inline docs/topics/multiplatform/multiplatform-hierarchy.md:220]: > It's a technical limitation of the template

It is true to some extent, but what I wanted to convey here is actually the opposite: it's not *only* a technical limitation, we actually *do not like* the idea of allowing this from the design perspective. Let's remove this sentence maybe?

File: docs/topics/multiplatform/multiplatform-hierarchy.md [hunk 3]
```diff
@@ -5,29 +5,25 @@ This means you can arrange a hierarchy of intermediate source sets for sharing t
...
+The template covers most popular cases. If you have a more advanced project, you can [configure it manually](#manual-configuration).
+It's a more low-level approach: it's more flexible, but requires more effort and knowledge. 
 
-Starting with Kotlin 1.9.20, Kotlin Gradle Plugin is shipped with the so-called "Default Source Sets Hierarchy Template"
-(or just "Default Hierarchy Template").
+## Default hierarchy template
 
-It's a [template](#see-the-full-hierarchy-template) that contains pre-defined intermediate source sets for some popular 
-cases. Kotlin Gradle Plugin will set up those source sets automatically based on the targets in your project
+The Kotlin Gradle plugin has a built-in default [hierarchy template](#see-the-full-hierarchy-template).
```
[the remark this answers is not in the document]
Comment (danil-pavlov) [reply docs/topics/multiplatform/multiplatform-hierarchy.md]: The documentation has to be relevant to the latest stable release, so we can't keep the note. But I added the Kotlin version to indicate when it became the default

File: docs/topics/multiplatform/multiplatform-hierarchy.md (hunk 1 above)
Comment (danil-pavlov) [reply docs/topics/multiplatform/multiplatform-hierarchy.md]: Changed to "Creating additional source sets"

File: docs/topics/multiplatform/multiplatform-hierarchy.md [hunk 4]
```diff
@@ -166,167 +106,146 @@ kotlin {
...
+    </tabs>
 
-#### Target shortcuts and ARM64 (Apple Silicon) simulators
+#### Modifying source sets
 
-The `ios`, `watchos`, and `tvos` target shortcuts don't include the simulator targets for ARM64 (Apple Silicon)
-platforms: `iosSimulatorArm64`, `watchosSimulatorArm64`, and `tvosSimulatorArm64`. If you use the target shortcuts
-and want to build the project for an Apple Silicon simulator, make the following adjustment to the build script:
+**Case**. You already have the source sets with the exact same names as those generated by the template but shared among
+different sets of targets in your project. For example, a `nativeMain` source set shared among not all the native
```
Comment (koshachy) [inline docs/topics/multiplatform/multiplatform-hierarchy.md:216]: Will it be better if we rephrase?

For example, a `nativeMain` source set is shared among only specific native targets: `linuxX64`, `mingwX64`, and `macosX64`.

File: docs/topics/whatsnew1530.md [hunk 5]
```diff
@@ -321,7 +321,7 @@ We've also introduced new targets that make Kotlin code run natively on Apple si
 
 They are available on both Intel-based and Apple silicon hosts. All existing targets are available on Apple silicon hosts as well.
 
-Note that in 1.5.30 we provide only basic support for Apple silicon targets in the `kotlin-multiplatform` Gradle plugin. Particularly, the new simulator targets aren't included in the [`ios`, `tvos`, and `watchos` target shortcuts](multiplatform-hierarchy.md#target-shortcuts). Learn how to [use Apple silicon targets with the target shortcuts](multiplatform-hierarchy.md#target-shortcuts-and-arm64-apple-silicon-simulators).
+Note that in 1.5.30 we provide only basic support for Apple silicon targets in the `kotlin-multiplatform` Gradle plugin. Particularly, the new simulator targets aren't included in the[`ios`, `tvos`, and `watchos` target shortcuts.
```
Comment (koshachy) [inline docs/topics/whatsnew1530.md:324]: Delete the square bracket "[".

**Pull request merged**